From 002e9b39baed9f7ef5423fd89417492366fda86a Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Tue, 6 Sep 2022 12:12:43 +0200
Subject: [PATCH 369/529] tee: optee: flag the irq chip to skip irq_set_wake

OPTEE asynchronous notifications doesn't support the set_wake option
currently. In case a device is using the optee irq and is marked as
wakeup source, this result in "Unbalanced IRQ xx wake disable"
backtrace, since:
- in irq_set_irq_wake(ON), wake_depth gets incremented, then reset due to
  set_irq_wake_real() returns an error (irq_set_wake() isn't unimplemented)
- in irq_set_irq_wake(OFF), wake_depth is always 0, hence the warning
So flag the OPTEE irq_chip to skip set_wake.

Change-Id: Ieea3acbd2aed15bcdc66fb21e8e71ef5f9b24aa3
Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
---
 drivers/tee/optee/smc_abi.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/tee/optee/smc_abi.c
+++ b/drivers/tee/optee/smc_abi.c
@@ -1048,6 +1048,7 @@ static struct irq_chip optee_it_irq_chip
 	.name = "optee-it",
 	.irq_disable = optee_it_irq_mask,
 	.irq_enable = optee_it_irq_unmask,
+	.flags = IRQCHIP_SKIP_SET_WAKE,
 };
 
 static int optee_it_alloc(struct irq_domain *d, unsigned int virq,
